### DISCUSSION TITLE
New to epd - fomer Kobo Hacker ;)

### DISCUSSION THREAD

**Discussion by noonasGit** [txoof/epd_display#72] [General]
Hello there!

I love what you have done here, and I may come back to develop a few plugins for your framework.
For now I am trying to figure out two things (I have an epd7in5 epaper screen (actually it is epd5in7b with red).
I have learned that you can use either driver (one obviously does not show red, but refresh is faster).
In your code I see that you can have short refresh and the default one - how do you do this ?
Also do you know how to print text in grey? My screen has 4 grey scales, but for now I cannot seem to access the grey?

thanks!

**Comment by noonasGit** [comment]
Found out how do to the grey text:     

Using "L" as mode and defining the RGB values for the colour of the text helps ;)

See snippet here: 

    black = 'rgb(0,0,0)'
    white = 'rgb(255,255,255)'
    grey = 'rgb(162,162,162)'

    # Drawing on the Horizontal image
    logging.info("1.Drawing on the Horizontal image...")
    Himage = Image.new('L', (epd.width, epd.height), 255)  # 255: clear the frame

**Reply by txoof** [reply]
@noonasGit 
Greetings! I'd love to collaborate on some more plugins. Though, I might suggest that you move over to [PaperPi](https://github.com/txoof/PaperPi). This version (epd_display) is no longer supported and linked against a much older version of the epdLib that runs this project.

As for the 7in5b panels, we support 1 bit mode on those, but not grayscale. EpdLib will do an approximation of grayscale using dithering. As you've found the 7in5 mode is much faster than the 7in5b because there's no update/refresh cycle for the red portion of the display. 

I wasn't aware that any of the non IT8951 models other than the epd3in7 supported grayscale. **Can you give me the exact model number you're working with?** I browsed the epd7in5b code and tried to find examples of grayscale, but couldn't put my finger on it.

The only models that I know of that definitely support grayscale are the IT8951 HD based panels and the epd3in7 (which is unsupported for this project). The IT8951 variants all support fast partial refresh as well, but are definitely more expensive.

**Comment by noonasGit** [comment]
Hello there, and thanks for replying.

Indeed, it was just dithering (not true greyscale)
I half heartedly, hoped that I would find a colour code that would be actual grey but after some hacking around and testing all the colours I realized it was just a "fake" grey via dithering.
I do have the red so there is that.

I kept reading about partial refresh (or fast refresh) but it seems that my screen (Waveshare epd7in5b_V2) does not support this either.

I am able to use the non-red version of the model epd7in5_V2 and it does refresh faster (since not addressing the red layer at all).

Having been spoiled with the Kobo's CPU and wonderful screen, the waveshare seems less fun ;) but it is clearer to read on the other hand.

Once I am done with my "dashboard" I will get back to you for some of my portions.
I have developed a Transit app client that can fetch live departure data for example

/Jonas

**Comment by txoof** [comment]
@noonasGit 

I'd love to add your transit app and dashboard as plugins if you're interested. Once you have the logic for the plugin ready, creating a plugin for it is just a matter of wrapping it in an update function and defining a layout. I can help with any part of that.

If you want a fast refresh rate or partial refresh, check out the IT8951 Screens. They're way nicer and can actually do gray scale.
